windows/devices/uwb/UwbConnector.cxx

```diff
@@ -23,72 +23,94 @@ namespace uwb
 {
 struct RegisteredCallbackToken
 {
+    /**
+     * @brief Construct a new Registered Callback Token object
+     *
+     * @param deregister the lambda that is passed by copy that handles deregistration.
+     * The lambda assumes that the caller is holding the mutex UwbConnector::m_eventCallbacksGate
+     */
+    RegisteredCallbackToken(std::function<void(RegisteredCallbackToken*)> deregister) :
+        Deregister([this, deregister]() {
+            return deregister(this);
+        })
+    {}
     virtual ~RegisteredCallbackToken() = default;
+
+    /**
+     * @brief Handles Deregistration
+     * The lambda assumes that the caller is holding the mutex UwbConnector::m_eventCallbacksGate
+     */
+    std::function<void()> Deregister;
 };
 
 struct RegisteredSessionCallbackToken : public RegisteredCallbackToken
 {
-    RegisteredSessionCallbackToken(uint32_t sessionId) :
-        SessionId(sessionId){};
+    RegisteredSessionCallbackToken(std::function<void(RegisteredCallbackToken*)> deregister, uint32_t sessionId) :
+        RegisteredCallbackToken(deregister), SessionId(sessionId){};

```

Comment:
Similar to prior comment, will be best to move `deregister`, same for all the other derived class c'tor init lists:

```suggestion
        RegisteredCallbackToken(std::move(deregister)), SessionId(sessionId){};
```